core/src/main/scala/org/apache/spark/SparkConf.scala

```
@@ -406,6 +406,15 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
   def getBoolean(key: String, defaultValue: Boolean): Boolean = {
     getOption(key).map(_.toBoolean).getOrElse(defaultValue)
   }
+  /** Get a parameter as a Option[Boolean] */
+  def getOptionBoolean(key: String): Option[Boolean] = {
```

You don't need these. Just have the defaults retain the prior behavior.

---

I was unsure about copying the defaults from the mesos protos(they were set to None earlier which mesos translates as defaults), and `FrameworkInfo.setCheckpoint()` expects an Option[Boolean] and there was no helper function that extracted Option[Boolean] from SparkConf

---

Maybe you can do this:
```scala
sc.conf.getOption("spark.mesos.checkpoint").map(_.toBoolean)
```

---

@xanec I drifted away from this because I wasn't sure about the behavior in case of non matching types(exception), but then noticed that `getDouble` and `getLong` also throw exception and do not default to the default value (bug or design ?).
So if they don't handle exceptions, then I can use what you suggested

---

Yeah, I think it is better to throw exception for non matching type rather than silently revert to default value. Could be bug prone during operation.